### ISSUE TITLE
Runs on localhost:19999 but not on ip:19999?

### ISSUE THREAD

**Issue by aapis** [firehol/netdata#1203]
What am I missing here?

```
ssh my-server
curl http://localhost:19999
# prints a bunch of HTML, clearly this is the dashboard
```

In the browser, both my-server.com:19999 and my-server-ip:19999 error out ("$browser cannot open the page").

Netdata 1.4.0, Ubuntu 16.04.

Update: possibly a better description of the problem?

Lets say my server IP is 1.1.1.2. The problem I'm facing is that netdata will run on 10.0.0.4:19999 and localhost:19999 but not 1.1.1.2:19999. I can curl these local addresses from the SSH session, but can't view the dashboard in a browser.

**Comment by vpashka** [comment]
It is right.
real ip != localhost (aka 127.0.0.1)
but netdata default run as "localhost" in /etc/netdata/netdata.conf   
bind to =

**Comment by aapis** [comment]
Sorry, not sure I understand your recommendation.  You're saying I should set `bind to = my-server-ip`?

**Comment by vpashka** [comment]
curl http://10.0.0.4:19999  
This command does not work?

**Comment by aapis** [comment]
It works, prints the same dashboard HTML stuff as `curl http://localhost:19999`.

**Comment by vpashka** [comment]
If you want the browser to work on hostname, it is necessary for it to be written to the file 
/etc/hosts

**Comment by vpashka** [comment]
Maybe I misunderstand what your problem is.
PS And yet my bad english :(

**Comment by aapis** [comment]
I haven't had to add anything to the hosts file on the other servers I've installed this software on.

Lets say my server IP is 1.1.1.2.  The problem I'm facing is that netdata will run on `10.0.0.4:19999` and `localhost:19999` but not `1.1.1.2:19999`.  I can curl these local addresses from the SSH session, but can't view the dashboard in a browser.  Hope that clarifies things a bit.

**Comment by vpashka** [comment]
If your server has a public IP address, then you just need to fit it into netdata settings. If you want to go to ssh to your server and at the same time look into browser. That maybe you should use
ssh LocalForward.

**Comment by ktsaou** [comment]
On the server you can't access, could you please run this:

```sh
netstat -nat | grep 19999
```

This is mine:

```sh
# netstat -nat | grep 19999
tcp        0      0 0.0.0.0:19999           0.0.0.0:*               LISTEN
tcp6       0      0 :::19999                :::*                    LISTEN
```

What is yours?

**Comment by aapis** [comment]
```
tcp        0      0 10.0.0.4:19999          0.0.0.0:*               LISTEN     
tcp        0      0 10.0.0.4:19999          10.0.0.4:52016          TIME_WAIT  
tcp6       0      0 :::19999                :::*                    LISTEN     
```

**Comment by ktsaou** [comment]
ok, so it only listens to `10.0.0.4` only.

Can you post this one too:

```sh
grep 'bind to' /etc/netdata/netdata.conf
```

This is mine, which makes netdata bind to all IPs of the system:

```sh
# grep 'bind to' /etc/netdata/netdata.conf
        bind to = *
```

In general you can leave it like that (an asterisk), or set it to one or more IPs:

```
    bind to = 10.0.0.4:19999 1.1.1.2:19999 127.0.0.1:19999
```

**Comment by aapis** [comment]
```shell
# $ grep 'bind to' /etc/netdata/netdata.conf
bind to = *
```

Previously it was commented out, presumably using `10.0.0.4` as a default.  I just changed it to `*` and restarted the service, no change.  Still able to `curl http://10.0.0.4:1999` and get the dashboard, unable to view in the browser.

**Comment by ktsaou** [comment]
hm...

Can you post this?

```sh
curl -s http://localhost:19999/netdata.conf | grep "bind to"
```

**Comment by mario-coin** [comment]
Have anyone solved this issue?
I checked the "bind to" option is set to all IPs of the system (asterisk).
```
#netstat -nat | grep 19999
tcp        0      0 0.0.0.0:19999           0.0.0.0:*               LISTEN
tcp6       0      0 :::19999                :::*                    LISTEN
```
```
#grep 'bind to' /etc/netdata/netdata.conf
        # bind to = *
```
```
curl -s http://localhost:19999/netdata.conf | grep "bind to"
        # bind to = *
```

**Comment by ktsaou** [comment]
Your netdata listens on all IPs.
Do you have a firewall installed?

**Comment by mario-coin** [comment]
Yep
```
ufw status | grep 19999
19999/tcp                  ALLOW       Anywhere
19999/tcp (v6)             ALLOW       Anywhere (v6)
```

**Comment by ktsaou** [comment]
Well... since accessing via localhost work and `netstat` reports that netdata has binded its socket properly, I can't think of anything related to netdata that can be wrong.

I would tcpdump the traffic to see if it reaches the server. If it does, I would try to see if the firewall really allows it (there should be a counter somewhere, there is a counter in `iptables -nxvL`).

**Comment by mario-coin** [comment]
I see.
Thanks for your fast feedback :-)

From iptables, the lines bellow tells me it's fine, right?
```
iptables -nxvL | grep 19999
       1       60 ACCEPT     tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:19999
```

Sorry, I'm not very familiarized with firewall permissions.
I'll keep digging.

**Comment by ktsaou** [comment]
This one says that there has been one connection accepted. But it might be the one from localhost.
When you connect from any other host, does it increment?

**Comment by mario-coin** [comment]
Could find neither localhost, nor 127.0.0.1, nor 0.0.0.0 related to iptables. There's no DROP command listed, and the only REJECT command would be to deny all requests anywhere, which I think it must be that way.
I've disabled UFW, just to be sure, and got no success.
I also tried applying curl to the id address from prompt (UFW running), and it works.

Sorry if I missed anything. I'm learning while trying to solve the problem.

**Comment by mario-coin** [comment]
The system had to restart something somehow.
Obviously it's not a good answer, but I have done nothing else that I've already done on the tutorial, which wasn't working.

Anyway, the system works fine now. Sorry for the trouble.

**Comment by ktsaou** [comment]
ok. This should not be necessary, but anyway it is nice you have solved it somehow.

**Comment by aapis** [comment]
@ktsaou This is still a problem for me.  Can you provide any other guidance?

**Comment by ktsaou** [comment]
hm...  is it possible that you have 2 servers with the same IP?

Could you please run on the server:

```
tcpdump -i any -n port 19999
```

and then try to access `http://your.server.ip:19999/`. Does it log anything?

**Comment by aapis** [comment]
Thanks for getting back to me!

No logs when I do that.

**Comment by ktsaou** [comment]
> No logs when I do that.

Then, I am afraid that the packets are not routed to this server...

Try to `tcpdump` port 22 and ssh to the server.  It should log. If it does, the `tcpdump` command is valid and you should also see logs on port 19999.

If the IP of the server is valid, you have another firewall somewhere blocking port 19999.

**Comment by aapis** [comment]
Yeah, I see logs when I `curl http://localhost:19999` from another SSH session but not when hitting it in the browser.  Ok, that narrows things down a bit.  Thanks for the help!

**Comment by aapis** [comment]
Yup, turns out there was a firewall that I didn't know about which only opened 22, 80 and 443 by default.  For anyone who comes here in the future, my server is hosted in the MS Azure cloud and this firewall is part of that service (apparently).

**Comment by ngethe** [comment]
I had the same issue the port were closed from AWS i open the port using this solution..  https://stackoverflow.com/questions/5004159/opening-port-80-ec2-amazon-web-services/10454688#10454688

**Comment by p-thurner** [comment]
When using the exact `bind to` line from the doc: https://docs.netdata.cloud/web/server/

`bind to = 127.0.0.1=dashboard 10.1.1.1:19998=management|netdata.conf hostname:19997=badges [::]:19996=streaming localhost:19995=registry *:http=dashboard unix:/tmp/netdata.sock`

I get this in the logs:

```
Jan 27 01:09:48 debian netdata: LISTENER: getaddrinfo('10.1.1.1', '19998=management|netdata.conf'): Servname not supported for ai_socktype
Jan 27 01:09:48 debian netdata[48191]: 2019-01-27 01:09:48: netdata ERROR : MAIN : LISTENER: getaddrinfo('10.1.1.1', '19998=management|netdata.conf'): Servname not supported for ai_socktype
Jan 27 01:09:48 debian netdata[48191]: 2019-01-27 01:09:48: netdata ERROR : MAIN : LISTENER: getaddrinfo('hostname', '19997=badges'): Servname not supported for ai_socktype
Jan 27 01:09:48 debian netdata: LISTENER: getaddrinfo('hostname', '19997=badges'): Servname not supported for ai_socktype
Jan 27 01:09:48 debian netdata[48191]: 2019-01-27 01:09:48: netdata ERROR : MAIN : LISTENER: getaddrinfo('::', '19996=streaming'): Servname not supported for ai_socktype
Jan 27 01:09:48 debian netdata[48191]: 2019-01-27 01:09:48: netdata ERROR : MAIN : LISTENER: getaddrinfo('localhost', '19995=registry'): Servname not supported for ai_socktype
Jan 27 01:09:48 debian netdata[48191]: 2019-01-27 01:09:48: netdata ERROR : MAIN : LISTENER: getaddrinfo('(null)', 'http=dashboard'): Servname not supported for ai_socktype
Jan 27 01:09:48 debian netdata: LISTENER: getaddrinfo('::', '19996=streaming'): Servname not supported for ai_socktype
Jan 27 01:09:48 debian netdata: LISTENER: getaddrinfo('localhost', '19995=registry'): Servname not supported for ai_socktype
Jan 27 01:09:48 debian netdata: LISTENER: getaddrinfo('(null)', 'http=dashboard'): Servname not supported for ai_socktype
```

I'm trying to get everything running on localhost, except accept streaming on port 12345. Would this be correct? It fails with the same error for the streaming part:

```
bind to = 127.0.0.1:19999 *:20000=streaming
```

**Comment by cakrit** [comment]
@p-thurner thanks for the report. The fix is #5275, you can get it and compile from source if you don't want to wait for the commit and the nightly build.

**Comment by niroma** [comment]
Hi,
Same problem for me ... netdata was working fine 'til today and now I can't access http:myip:19999 anymore ... connection refused 
I'm running netdata on ubuntu 18.04 and even if I disable ufw problem is still here (service is working fine, and i got content running curl 127.0.0.1:19999)
I tried to set  bind to = * in netdata.conf but still doesn't work :'(

**Comment by ilyam8** [comment]
@niroma 	

please do `sudo ss -ntlp | grep netdata`

**Comment by cakrit** [comment]
Also, what was your `bind to` before you changed it to `*`?

**Comment by niroma** [comment]
Hi guys, 
Thank you for the fast replies :)
@ilyam8 
LISTEN   0         128               127.0.0.1:8125             0.0.0.0:*        users:(("netdata",pid=13186,fd=17))
LISTEN   0         128                 0.0.0.0:19999            0.0.0.0:*        users:(("netdata",pid=13186,fd=3))
LISTEN   0         128                   [::1]:8125                [::]:*        users:(("netdata",pid=13186,fd=16))
LISTEN   0         128                    [::]:19999               [::]:*        users:(("netdata",pid=13186,fd=4))
@cakrit 
Today it was bin to = 127.0.0.1 
Don't what it was yesterday as everything was working fine ;)

BTW thank you again :)

**Comment by cakrit** [comment]
I replicated it. Lemme see what's happening...

**Comment by niroma** [comment]
@cakrit just edited my previous answer with "netstat -nat | grep 19999" if this can help :)

**Comment by cakrit** [comment]
Nope, I didn't replicate it. It was just the SSL redirect that messed with my request.

**Comment by niroma** [comment]
@cakrit thanks for your time, nevermind as its working by now for me !
OFF TOPIC : I'm running litespeed and I'd love to proxy/password protect netdata ... Any clues about how to do this ? As docs aren't really litespeed compliant :/

**Comment by cakrit** [comment]
With `bind to = 127.0.0.1` I have the following:
```
LISTEN    0         128              127.0.0.1:8125             0.0.0.0:*        users:(("netdata",pid=24045,fd=32))                                            
LISTEN    0         128              127.0.0.1:19999            0.0.0.0:*        users:(("netdata",pid=24045,fd=3))                                             
LISTEN    0         128                  [::1]:8125                [::]:*        users:(("netdata",pid=24045,fd=31))  
```
With `bind to = 127.0.0.1` I have the following:
```
LISTEN    0         128              127.0.0.1:8125             0.0.0.0:*        users:(("netdata",pid=29266,fd=33),("netdata",pid=28737,fd=33))                
LISTEN    0         128                0.0.0.0:19999            0.0.0.0:*        users:(("netdata",pid=29266,fd=3),("netdata",pid=28737,fd=3))                  
LISTEN    0         128                  [::1]:8125                [::]:*        users:(("netdata",pid=29266,fd=32),("netdata",pid=28737,fd=32))                
LISTEN    0         128                   [::]:19999               [::]:*        users:(("netdata",pid=29266,fd=4),("netdata",pid=28737,fd=4))  
```

Regardless of the setting, http://127.0.0.1:19999 works. 

What's your netdata version? 
I'll need to go off now, but we'll keep investigating, this is strange.

**Comment by niroma** [comment]
@cakrit hey man thanks again for watchning
NetData is auto updating, i'm currently running v1.15.0-53-nightly

**Comment by ilyam8** [comment]
> its working by now for me

I see the problem is solved.

> OFF TOPIC : I'm running litespeed and I'd love to proxy / ssl redirect / password protect netdata ... Any clues about how to do this ? As doc isn't really litespeed compliant :/

i rememeber it right `litespeed` thing is not free, i suggest to ask their support/ on their forum.

In case you are not aware - https://docs.netdata.cloud/web/server/#enabling-tls-support

**Comment by cakrit** [comment]
Tried it on Ubuntu 18.0.4 as well, can't replicate it. @niroma, if you can get it to happen again, please send me your netdata.conf, access.log and error.log to chris@netdata.cloud (just in case it contains sensitive info).

**Comment by kklepper** [comment]
Just in case someone has the trouble I had:

My server runs Ubuntu 20.x and I obviously want to access netdata from a browser, here Windows/Opera. curl from the terminal is fine, but Opera times out. 

After some research I thought the Windows Firewall would be the problem, so I opened port 19999 manually, both ingoing and outgoing. 

Still timeout. What now?

Wrong perspective, wrong jobsite. It was my server blocking. 

Solution: open firewall port 19999 on server -- obvious, if I think about it. The above Netdata Community link opened my eyes.

**Comment by ktsaou** [comment]
Thank you for reporting this @kklepper 

@Ferroin I see ufw supports application profiles https://askubuntu.com/questions/409013/how-do-you-create-an-app-profile-for-ufw

Possibly, we can ship such an app profile for ufw as `/etc/ufw/applications.d/netdata`:

```
[netdata]
title = "Netdata real-time performance monitoring web server"
description = "Netdata agent Web UI and Streaming https://www.netdata.cloud"
ports = 19999/tcp

[netdata-statsd]
title = "Netdata statsd server for APM metrics"
description = "Netdata statsd server https://learn.netdata.cloud/docs/agent/collectors/statsd.plugin"
ports = 8125/tcp|8125/udp
```

Then users can do this:

```sh
ufw allow netdata
```

**Comment by Ferroin** [comment]
@ktsaou Indeed, this has been something I’d been toying with doing for a while now, I just haven’t had the time. We probably would also want an equivalent for firewalld (the other popular option for this type of thing).

I’ll create an issue for this.

**Comment by ktsaou** [comment]
@Ferroin you rock! Yes, please do that!

**Comment by minto3792** [comment]
Thanks for your helpful comments! Here's how I fixed this issue on my Pop!_OS 22.04 system:

1. Edit /etc/netdata/netdata.conf using sudo nano
2. Change bind socket to IP = 127.0.0.1 to bind socket to IP = 192.168.1.20 (replace with your machine's IP)
3. Open up the Netdata firewall port (19999) for remote access

That's it! You should now be able to access Netdata remotely.
